src/util/merge_irep.cpp

```diff
@@ -145,18 +145,26 @@ void merge_irept::operator()(irept &irep)
 
 const irept &merge_irept::merged(const irept &irep)
 {
+  if(irep.get_sub().empty() && irep.get_named_sub().empty())
+    return *irep_store.insert(irep).first;
+
   irep_storet::const_iterator entry=irep_store.find(irep);
   if(entry!=irep_store.end())
     return *entry;
 
+  bool same_irep = true;
   irept new_irep(irep.id());
```

Comment:
How about starting with `new_irep = irep` (cheap refcounted copy), then selectively overwriting `sub` / `named_sub`? That way we only make a CoW break if `merged` does some merging and we actually update either `sub` or `named_sub`, whereas at present we'll build and then discard `new_irep` in that case.

Alternatively if you take my suggestion above then `new_irep` will just be `irept &new_irep = *irep_store.insert(irep).first`

Reply:
I've got a follow-up change that will avoid discarding `new_irep` (well, in fact it will never build `new_irep` in the first place), but I wanted to benchmark these independently (as they optimise on different aspects, this here reduces the number of `operator==` calls, while the other will reduce the number of `detach` calls). Let's look into this on that other PR (yet to come).